File: 4-kyu/strip-comments/strip-comments.py
def strip_comments(text, markers):
    lines = text.split('\n')
    result = []
​
    for line in lines:
        for marker in markers:
            if marker in line:
                line = line.split(marker)[0]
        result.append(line.rstrip())
​
    return '\n'.join(result)

File: 4-kyu/strip-comments/strip-comments.js
function solution(text, markers) {
  return text.split('\n').map(line => {
    const cutIndex = markers
      .map(marker => line.indexOf(marker))
      .filter(i => i !== -1);
    
    const minIndex = Math.min(...cutIndex, line.length);
    return line.slice(0, minIndex).trimEnd();
  }).join('\n');
}